pkg/kapp/resourcesmisc/apps_v1_deployment.go

```diff
@@ -59,6 +59,12 @@ func (s AppsV1Deployment) IsDoneApplying() DoneApplyState {
 				return DoneApplyState{Done: true, Successful: false, Message: fmt.Sprintf(
 					"Deployment failed to delete pods: %s (message: %s)", cond.Reason, cond.Message)}
 			}
+
+		case "FailedCreate":
+			if cond.Status == corev1.ConditionTrue {
+				return DoneApplyState{Done: true, Successful: false, Message: fmt.Sprintf(
```

Comment:
We don't want to mark it as done in this case as it could still succeed in some scenarios. Dmitriy also mentioned the same in the issue:
> i do not think we need to consider progress as done though, in this case